src/main/groovy/wooga/gradle/github/base/internal/RepositoryInfo.groovy

```diff
@@ -0,0 +1,38 @@
+package wooga.gradle.github.base.internal
+
+
+import org.ajoberstar.grgit.Grgit
+import org.gradle.api.Project
+import org.gradle.api.provider.Provider
+
+class RepositoryInfo {
+
+    private static final String DOMAIN = "github.com"
+    private static final String DEFAULT_REMOTE = "origin"
+
+    final Project project
+    final Grgit git
+
+    RepositoryInfo(Project project, Grgit git) {
+        this.project = project
+        this.git = git
+    }
+
+    Provider<String> repositoryNameProviderFromLocalGit() {

```

Comment:
There is no need here to use the term `provider` inside the method name. I also thing that a `getter` would fit better.

```suggestion
    Provider<String> getRepositoryNameFromLocalGit() {

```

Reply:
I'll include that in the upcoming commit. Won't use the suggestion here to avoid potential merging issues.